File: app/src/main/java/com/example/footmatch/ClasificacionAdapter.java
package com.example.footmatch;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;


import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.footmatch.modelo.Equipo;
import com.example.footmatch.modelo.Partido;


import java.util.List;


public class ClasificacionAdapter extends RecyclerView.Adapter<ClasificacionAdapter.ClasificacionViewHolder> {
    public interface OnItemClickListener {
        void onItemClick(Equipo item);
    }
    public List<Equipo> equipos;
    //Se utilizará mas adelante para ir a la pantalla de equipos
    private final OnItemClickListener listener;

    public ClasificacionAdapter(List<Equipo> equipos, OnItemClickListener listener){
        this.equipos = equipos;
        this.listener = listener;
    }



    @Override
    public int getItemCount() {
        return equipos.size();
    }


    public static class ClasificacionViewHolder extends RecyclerView.ViewHolder{

        private ImageView escudo;
        private TextView nombreEquipo;
        private  TextView puntosEquipo;

        public ClasificacionViewHolder(View itemView) {
            super(itemView);
            escudo = (ImageView)itemView.findViewById(R.id.imagenEquipo);
            nombreEquipo = (TextView) itemView.findViewById(R.id.equipoLiga);
            puntosEquipo = (TextView) itemView.findViewById(R.id.puntosEquipo);

        }

        // asignar valores a los componentes
        public void bindUser(@NonNull final Equipo equipo, final OnItemClickListener listener) {
            nombreEquipo.setText(equipo.getNombre());
            puntosEquipo.setText(String.valueOf(equipo.getPuntos()));
            //Picasso.get()
            //        .load(equipo.getUrlImagenEscudo()).into(escudo);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override public void onClick(View v) {
                    // Metodo para accerder a la plantilla desde clasificacion
                }
            });
        }
    }




    @NonNull
    @Override
    public ClasificacionViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.linear_clasification_equipo, parent, false);
        return new ClasificacionViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull ClasificacionViewHolder holder, int position) {
        Equipo equipo= equipos.get(position);
        holder.bindUser(equipo, listener);
    }


}

File: app/src/main/java/com/example/footmatch/ClasificacionAdapter.kt
package com.example.footmatch

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import coil.load
import com.example.footmatch.ClasificacionAdapter.ClasificacionViewHolder
import com.example.footmatch.modelo.pojos.clasificacion.StandingsResult
import com.example.footmatch.modelo.pojos.clasificacion.Table
import com.example.footmatch.util.images.SvgLoader.Companion.loadUrl

class ClasificacionAdapter(
    var equipos: StandingsResult, //Se utilizará mas adelante para ir a la pantalla de equipos
    val onItemSelected: (String) -> Unit
) : RecyclerView.Adapter<ClasificacionViewHolder>() {

    override fun getItemCount(): Int {
        return equipos.standings[0].table.size
    }

    class ClasificacionViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private val nombreEquipo: ImageView
        private val puntosEquipo: TextView
        private val posicion: TextView
        private val golesAFavor: TextView
        private val golesEnContra: TextView
        private val diferenciaGoles: TextView
        private val ganados: TextView
        private val empatados: TextView
        private val perdidos: TextView

        init {
            ganados = itemView.findViewById<View>(R.id.partidosGanados) as TextView
            perdidos = itemView.findViewById<View>(R.id.partidosPerdidos) as TextView
            empatados = itemView.findViewById<View>(R.id.partidosEmpatados) as TextView
            nombreEquipo = itemView.findViewById<View>(R.id.logoEquipo) as ImageView
            puntosEquipo = itemView.findViewById<View>(R.id.puntosEquipo) as TextView
            posicion = itemView.findViewById<View>(R.id.posClasificacion) as TextView
            golesAFavor = itemView.findViewById<View>(R.id.golesFavor) as TextView
            golesEnContra = itemView.findViewById<View>(R.id.golesContra) as TextView
            diferenciaGoles = itemView.findViewById<View>(R.id.diferenciaGoles) as TextView
        }

        // asignar valores a los componentes
        fun bindUser(equipo: Table, onItemSelected: (String) -> Unit) {
            if (equipo.team.crest == null) {
                // cargar imagen visitante por defecto
                nombreEquipo.load(R.string.teamDefaultLogo.toString())
            }else{
                val isSvg = equipo.team.crest.endsWith("svg",ignoreCase = true)
                if (isSvg){
                    nombreEquipo.loadUrl(equipo.team.crest)
                }else{
                    nombreEquipo.load(equipo.team.crest)
                }
            }
            puntosEquipo.text = equipo.points.toString()
            posicion.text = equipo.position.toString()
            golesAFavor.text = equipo.goalsFor.toString()
            golesEnContra.text = equipo.goalsAgainst.toString()
            diferenciaGoles.text= equipo.goalDifference.toString()
            ganados.text = equipo.won.toString()
            perdidos.text = equipo.lost.toString()
            empatados.text = equipo.draw.toString()

            itemView.setOnClickListener {
                onItemSelected(equipo.team.id.toString())
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ClasificacionViewHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.linear_clasification_equipo, parent, false)
        return ClasificacionViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: ClasificacionViewHolder, position: Int) {
        val equipo: Table = equipos.standings[0].table[position]
        holder.bindUser(equipo, onItemSelected)
    }


}